presto-main/src/main/java/com/facebook/presto/server/QuerySessionSupplier.java

```diff
@@ -178,7 +178,12 @@ public Session createSession(QueryId queryId, SessionContext context, Optional<S
             sessionBuilder.setClientTransactionSupport();
         }
 
-        sessionBuilder.setPath(new SqlPath(path));
+        if (path.isPresent()) {
+            sessionBuilder.setPath(new SqlPath(path.get()));
+        }
+        else {
+            sessionBuilder.setPath(new SqlPath(null));
```

Comment:
Should this be changed to use Optional instead of null? This way you can separate a null check from an empty path.